test/integration/targets/sysctl/tasks/main.yml

```diff
@@ -143,31 +168,80 @@
     that:
       - sysctl_test3 is changed
       - 'sysctl_check3.stdout_lines == ["net.ipv4.ip_forward = 1"]'
+  when: sysctl_cmd_supported
 
 - name: Try sysctl with no name
   sysctl:
     name:
     value: 1
     sysctl_set: yes
   ignore_errors: True
+  when: sysctl_cmd_supported
   register: sysctl_no_name
 
 - name: validate nameless results
   assert:
     that:
       - sysctl_no_name is failed
       - "sysctl_no_name.msg == 'name cannot be None'"
+  when: sysctl_cmd_supported
 
 - name: Try sysctl with no value
   sysctl:
     name: Foo
     value:
     sysctl_set: yes
   ignore_errors: True
+  when: sysctl_cmd_supported
   register: sysctl_no_value
 
 - name: validate nameless results
   assert:
     that:
       - sysctl_no_value is failed
       - "sysctl_no_value.msg == 'value cannot be None'"
+  when: sysctl_cmd_supported
+
+## 
+## sysctl - sysctl_set with ignore_unknown_keys and ignoreerrors (to ensure backward compatibility) 
+## 
+
+- name: set value for fictional key with ignore_unknown_keys
+  sysctl: 
+    name: k.doesnotexist
+    value: 1 
+    ignore_unknown_keys: yes
+    reload: "{{ sysctl_cmd_supported }}"
+  register: sysctl_ignore_unknown_keys0
+  when: ansible_os_family != 'RedHat'
```

Comment:
This results in skipping the tests on the only Linux VM we use in CI, which makes me think this is simply covering up an issue. To make sure this is correct we need to understand why it's failing.

In the case of our Docker containers we know it's a permissions issue, but that shouldn't be the case on our RHEL virtual machines.

Reply:
Totally agree and this is the part that does not really satisfy me. The test works fine on RedHat VMs and just fails on Docker Containers and (and that's the actual challenge I think) the response from the sysctl command in this case is not "key does not exist". Instead it was either read-only fs or something similiar. So the `sysctl -e` does not recognize it a invalid key and fails instead. 

I do not have an idea right now how to fix this or write a reliable and satisfying test here but I'm open for any input.